### DISCUSSION TITLE
Behavior of accepting invitations in certain states

### DISCUSSION THREAD

**Discussion by myOmikron** [hopfenspace/runciv#23] [General]
How should we handle accepting an invitation to a lobby from a user that isn't in the lobby anymore?

**Comment by CrsiX** [comment]
I can think of three different ways to handle this:

1. Only the lobby owner may invite other people. If the lobby owner leaves the lobby, the lobby gets closed automatically (and all members will be kicked), therefore all invitations become implicitly or explicitly unusable.
2. Every member may invite other people. If that user leaves the lobby, the invitation will be rejected/becomes unusable.
3. Every member may invite other people. If that user leaves the lobby, nothing changes and the invitation is still valid.

Personally, I think option 2 is the best option, with option 1 being the fallback. Option 2 makes it easily possible to play with a friend who in turn invites his friend, which you didn't know before / had no friendship with. Keeping the invitation active seems wrong IMHO.

**Comment by CrsiX** [comment]
So, I think we settled for option 2, if you don't have any other opinion. I would then create an issue to add an API endpoint for it, @myOmikron.
